nbextensions/usability/hide_input/main.js

```diff
@@ -1,51 +1,44 @@
 // Adds a button to hide the input part of the currently selected cells
 
-// Prevent this script from cluttering the IPython namespace
-(function (IPython) {
+define([
+    'jquery',
+    'base/js/namespace'
+], function(
+    $,
+    IPython
+) {
     "use strict";
 
-
-    var hide_input = function () {
-
+    var toggle_selected_input = function () {
         // Find the selected cell
         var cell = IPython.notebook.get_selected_cell();
         // Toggle visibility of the input div
-        cell.element.find("div.input").toggle('slow')
-
-        if ( cell.metadata.input_collapsed ) {
-            cell.metadata.input_collapsed = false;
-        } else {
-            cell.metadata.input_collapsed = true;
-        }
+        cell.element.find("div.input").toggle('slow');
+        cell.metadata.input_collapsed = ! cell.metadata.input_collapsed;
```

Comment:
I think we should use the same metadata structure `cell.metadata.hide_input` as in the other extensions (hide_input_all, runtools)